includes/forum-profile.php

```diff
@@ -176,6 +201,33 @@ public function get_post_history_by_user($user_id, $limit = false) {
         }
     }
 
+    public function get_edit_history_by_user($user_id, $limit = false, $post_id = null) {
+        $query       = '';
+        $query_limit = '';
+    
+        if ($limit) {
+            $elements_maximum = 50;
+            $elements_start   = $this->asgarosforum->current_page * $elements_maximum;
+    
+            $query_limit = "LIMIT {$elements_start}, {$elements_maximum}";
+        }
+    
+        $conditions = "WHERE p.author_id = %d";
+    
+        if ($post_id !== null) {
+            $conditions .= " AND h.post_id = %d";
+        }
+    
+        $query = "SELECT p.id, h.edited_content, h.edit_timestamp FROM {$this->asgarosforum->tables->edit_history} AS h
+                  LEFT JOIN {$this->asgarosforum->tables->posts} AS p ON h.post_id = p.id
+                  {$conditions} ORDER BY h.edit_timestamp DESC {$query_limit};";
+    
+        $prepared_query = $this->asgarosforum->db->prepare($query, $user_id, $post_id);
```

Comment:
is it safe to give prepare() 3 arguments with only 1 condition? won't prepare complain that it got too many arguments? maybe
```suggestion
        $prepare_arguments=aray($query,$use_id);
        if ($post_id !== null){
            $prepare_arguments[]=$post_id;
        )
        $prepared_query = $this->asgarosforum->db->prepare(...$prepare_arguments);